agent/structs/discovery_chain.go

```diff
@@ -191,6 +191,13 @@ type DiscoveryRoute struct {
 
 // compiled form of ServiceSplit
 type DiscoverySplit struct {
+	Definition *ServiceSplit `json:",omitempty"`
+	// Weight is not necessarily a duplicate of Definition.Weight since when
+	// multiple splits are compiled down to a single set of splits the effective
+	// weight of a split leg might not be the same as in the original definition.
+	// Proxies should use this compiled weight. The Definition is provided above
+	// for any other significant configuration that the proxy might need to apply
+	// to that leg of the split.
```

Comment:
Adding the definition is needed for the new config attached to split legs to propagate to xDS. It will also mean that the `/v1/discovery-chain` API will inherit this new data (which is good if anyone actually uses it). I've put a TODO to update this in API docs.